rules/mockery-to-prophecy/src/Rector/MethodCall/MockeryCreateMockToProphizeRector.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\MockeryToProphecy\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\ClassMethod;
+use Rector\Core\Rector\AbstractPHPUnitRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\MockeryToProphecy\MockeryUtilsTrait;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\MockeryToProphecy\Tests\Rector\MethodCall\MockeryToProphecyRector\MockeryToProphecyRectorTest
+ */
+final class MockeryCreateMockToProphizeRector extends AbstractPHPUnitRector
+{
+    use MockeryUtilsTrait;
+
+    /**
+     * @var array<string, class-string>
+     */
+    private $mockVariableTypesByNames;
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isInTestClass($node)) {
+            return null;
+        }
+
+        if ($node instanceof ClassMethod) {
+            $this->replaceMockCreationsAndCollectVariableNames($node);
+            $this->revealMockArguments($node);
+        }
+
+        return $node;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Changes mockery mock creation to Prophesize',
+            [
+                new CodeSample(
+                    '$mock = \Mockery::mock(\'MyClass\');
+        $service = new Service();
+        $service->injectDependency($mock);',
+                    '$mock = $this->prophesize(\'MyClass\');
+
+        $service = new Service();
+        $service->injectDependency($mock->reveal());'
+                ),
```

Comment:
This will eventually lead into random flying spaces everywhere. Mainly in generated docs.

Better here/nowdoc format:

https://github.com/rectorphp/rector/blob/69691157bdb0a0a27e8172925dd2bd7922e3bb29/rules/php74/src/Rector/Property/TypedPropertyRector.php#L79-L96